### PR TITLE
fix(framework): cast torch.Tensor only

### PR DESCRIPTION
**Pull request by bojiang**
## What does this PR address?

Fixes #(issue)

## Before submitting:

- [ ] Does the Pull Request follow [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary) naming? Here are [GitHub's
guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on how to create a pull request.
- [ ] Does the code follow BentoML's code style, both `make format` and `make lint` script have passed ([instructions](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#style-check-auto-formatting-type-checking))?
- [ ] Did you read through [contribution guidelines](https://github.com/bentoml/BentoML/blob/main/CONTRIBUTING.md#ways-to-contribute) and follow [development guidelines](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#start-developing)?
- [ ] Did your changes require updates to the documentation? Have you updated
  those accordingly? Here are [documentation guidelines](https://github.com/bentoml/BentoML/tree/main/docs) and [tips on writting docs](https://github.com/bentoml/BentoML/tree/main/docs#writing-documentation).
- [ ] Did you write tests to cover your changes?

## Who can help review?

Feel free to tag members/contributors who can help review your PR.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -64,7 +64,7 @@ install_requires =
     schema
     simple-di>=0.1.4
     starlette
-    typing-extensions>=4.0
+    typing-extensions>=3.7,<4
```
Comment (aarnphm) [inline setup.cfg:67]: Why do we set an upper lock version for typing-extension?

File: bentoml/_internal/frameworks/common/pytorch.py [hunk 2]
```diff
@@ -95,8 +95,9 @@ def _mapping(
                 return torch.Tensor(item, device=self.device_id)
             if LazyType["ext.PdDataFrame"]("pandas.DataFrame").isinstance(item):
                 return torch.Tensor(item.to_numpy(), device=self.device_id)
-            else:
-                return item.to(self.device_id)  # type: ignore # the overhead is trivial if it is already on the right device
+            if LazyType["torch.Tensor"]("torch.Tensor").isinstance(item):
+                return item.to(self.device_id)
+            return item  # type: ignore # the overhead is trivial if it is already on the right device
```
Comment (aarnphm) [inline bentoml/_internal/frameworks/common/pytorch.py:100]:
```suggestion
                return item.to(self.device_id)  # type: ignore # the overhead is trivial if it is already on the right device
            return item
```

File: setup.cfg (hunk 1 above)
Comment (bojiang) [reply setup.cfg:67]: read the doc

Comment (aarnphm) [reply setup.cfg:67]: no need to lock it

Comment (aarnphm) [reply setup.cfg:67]: I think we have gone through this b4.

Comment (aarnphm) [inline setup.cfg:67]:
```suggestion
    typing-extensions>=4.0
```

**Review by aarnphm — APPROVED** [review]

**Comment by aarnphm** [comment]
address in #2857

**Pull request closed without merging**